### ISSUE TITLE
revise project goals and non goals

### ISSUE THREAD

**Issue by melvinzhang** [magarena/magarena#372]
@lodici brought up a good point about project goals in https://github.com/magarena/magarena/commit/9afc70e12c3af2c249a16e761533b7bae6a35f22#commitcomment-11751524
The relevant part of the comment is:
> I think we all agree we don't want to end up with a Forge II. And I would rather not spend time working on stuff that is incidental to the main goals of Magarena (whatever they are, exactly!).

I've mostly inherited my views from the original project started by ubeefx and based on comments from users of the early days on what makes this project special. They are:
* Quick and easy to get in a game when you have a short break
* AI is challenging
* UI is simple and easy to use

Some implications from the early goals:
* focus on a small set of cards that the AI plays well (keeps it challenging)
* modify the game rules to suit the AI (patch up AI weakness via game rules changes)
* follow the limited format and use randomly generated decks so that both sides are more evenly matched, as the AI plays simple creature based strategies better than combo or control.
* use randomly generated decks so that it is quick and easy to start a new game, players just select the colors they want to play

Some of the non-goals:
* add cards that the AI does not use properly (this reduces the challenge)
* support human vs human play (this makes the AI redundant, our focus is on single player)

We have moved on somewhat since then it is a good time to revise our goals and also non goals moving forward.

Looking at the state of the related projects in this field, most projects either focus on human vs human play (XMage) or on a large card pool (Forge). I feel that our focus on AI is what makes us unique, together with our clean and simple UI.

The original goal of "Quick and easy to get in a game when you have a short break" probably needs to be revised. There are many different ways players want to play the game, for example many players have suggested the idea of a campaign mode with deck building as a core component.

**Comment by melvinzhang** [comment]
Testimonial from a user who gets the random deck gameplay: http://www.slightlymagic.net/forum/viewtopic.php?f=82&t=16575&p=171038&hilit=magarena#p170941

**Comment by ShawnieBoy** [comment]
He definitely gets the spirit of that format - Which does get me thinking. What we're presenting with Magarena's version of a Limited format is actually a new format entirely. It doesn't really fit with current limited formats, where the emphasis is on deckbuilding from a limited pool.

What we have is essentially a 'random' 40 card highlander. (It's not entirely random as there are some rules which take into account playability). Which I quite like playing 'blind' - as in not looking at the deck contents before a match. This format moves away from deckbuilding and puts more emphasis on player skill.

**Comment by lodici** [comment]
> VladVoivode explains the random deck gameplay better than I ever could...

Interesting. Thinking outside the box but how about a rogue-like mode of play :-

- the aim is to see how long you can survive given a random deck.
- define the pool of cards from which initial random decks will be generated. This could be quite sophisticated using something like the explorer filter or for quick play you could have a number of predefined pools maybe based on player level (novice, casual, nerd, etc).
- players can re-roll deck as many times as they like before starting the duel but otherwise cannot influence the cards in the deck.
- play for rewards. For example, increased health, ante card, dumber ai, view opponents deck/hand, etc.

**Comment by ShawnieBoy** [comment]
@lodici An interesting version of a campaign mode. I like it. Maybe only having an option to save the deck once you've lost.

Returning to the goals - One of the early effects of the 'quick games', with not having particular cards involved, I think is probably no longer relevant. With added support of Firemind and increased card pools, Constructed influence has modified this.

One of those early implications of 'quick games' which I'd like to see changed, is the added possibility of a game ending in a draw. Currently there can only be one winner and one loser. If the AI would view a draw as a loss, this could prevent degenerate play on the AI's part. Draws generally occur out of your control (triggered effects etc.) and losing to something out of your control, which would normally be a draw can be disheartening. There's also at least one card I can think of which creates a draw.

**Comment by melvinzhang** [comment]
> with not having particular cards involved, I think is probably no longer relevant

The negative consequence is that random decks are much weaker than before and are not as exciting to play, as the general power level is lower due to the inclusion of many cards diluting the card pool. See release/Magarena/mods/ubeefx_cube.txt for what the original card pool looks like.

I find that the larger formats like standard, modern are just not as suitable for random deck generation as a custom built cube.

**Comment by ShawnieBoy** [comment]
The power level of Limited formats are low. Those large formats are never used in Limited formats, they are always Block formats of some kind.

**Comment by ShawnieBoy** [comment]
Having cubes will still be an options, as they are a variant of Draft anway, but I can see there will (is) a change in the feel. I personally think it's for the better, as you have to evaluate cards in a different way than Constructed, and have to find creative uses for cards that you would normally not use in a constructed deck.

**Comment by lodici** [comment]
I think the question is still relevant - is there a need for the deck editor function from the main menu? My feeling is that it does not fit in with the quick play, random deck vibe. At the very least I don't think it should be available from the main menu which should focus on getting the player up and running a game a quickly as possible.

**Comment by ShawnieBoy** [comment]
>The negative consequence is that random decks are much weaker than before and are not as exciting to play, as the general power level is lower due to the inclusion of many cards diluting the card pool.

Just had a thought about this - There could be an option, or as default, to only use cards over a particular rating. Thereby eliminating all 'poor' cards if the player wishes. e.g. 3 star + decks

**Comment by ShawnieBoy** [comment]
One goal that's been ongoing - making all cards scriptable.

**Comment by melvinzhang** [comment]
Action: summarize the goals/non goals and write it into a wiki page and then close this issue.

**Comment by bilbo2** [comment]
Well, different people might want to use magarena differently ... some want quick play, some want rogue-style-quest play ... I personally use it to test real decks (give AI one of my decks or some new deck that pops out on deckstats and then try to play with/against it to see if it is strong or rubbish), so while some of my personal goals are the same (keep AI strong, as if it is weak like in forge, then it is almost useless for this), some are not (I'd like the supported card pool to be as large as possible, as from almost every deck and I'd like the game rules to deviate from real MtG as little as possible).

The random generator could be made better by taking into account some combos and if one card of a combo would be put into deck, the second has higher chance of getting in.
